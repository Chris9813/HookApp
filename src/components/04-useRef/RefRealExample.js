import React, { useState } from 'react'

import "../02-useEffect/effects.css"
import { MultipleCustomHooks } from '../03-examples/MultipleCustomHooks'

export const RefRealExample = () => {
    const [show, setshow] = useState(false)
    
    return (
        <div>
            <h1>Real Example Ref</h1>
            <hr/>
            {show && <MultipleCustomHooks />}
            <button
            className = "btn btn-primary mt-5"
            onClick = {() => {
                setshow(!show)
            }}
            > MIRA/OCULTA EL CONTENIDO, PERRA!!
            </button>
            
        </div>
    )
}
