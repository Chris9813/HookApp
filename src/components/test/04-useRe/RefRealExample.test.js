import { shallow } from 'enzyme'
import React from 'react'
import { RefRealExample} from '../../04-useRef/RefRealExample'

describe("pruebas en RefRealExample", () => {

    const wrapper = shallow(<RefRealExample />)

    test('debe mostrrse correctamente ', () => {
        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find("MultipleCustomHooks").exists()).toBe(false)
    })
    
    test('debe mostrar la info', () => {
        wrapper.find("button").simulate("click")
        expect( wrapper.find("MultipleCustomHooks").exists()).toBe(true)
    })
    
/*
    const wrapper =  shallow(<RefRealExample />)
    test('debe mostrarse correctamente ', () => {
        expect( wrapper ).toMatchSnapshot();
        expect(wrapper.find("MultipleCustomHooks").exists()).toBe(false)
    })

    test('debe mostrar el componente  MultipleCustomHooks', () => {
        wrapper.find("button").simulate("click")
        expect(wrapper.find("MultipleCustomHooks").exists()).toBe(true)
    })
*/

}) 
