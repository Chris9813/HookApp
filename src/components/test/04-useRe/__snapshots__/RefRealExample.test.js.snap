// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en RefRealExample debe mostrrse correctamente  1`] = `
<div>
  <h1>
    Real Example Ref
  </h1>
  <hr />
  <button
    className="btn btn-primary mt-5"
    onClick={[Function]}
  >
     MIRA/OCULTA EL CONTENIDO, PERRA!!
  </button>
</div>
`;
